src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Security/CryptographicHardwareIntrinsicsTests.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Testing;
+using Test.Utilities;
+using Xunit;
+using VerifyCS = Test.Utilities.CSharpCodeFixVerifier<
+    Microsoft.NetCore.Analyzers.Security.CryptographicHardwareIntrinsicsAnalyzer,
+    Microsoft.CodeAnalysis.Testing.EmptyCodeFixProvider>;
+using VerifyVB = Test.Utilities.VisualBasicSecurityCodeFixVerifier<
+    Microsoft.NetCore.Analyzers.Security.CryptographicHardwareIntrinsicsAnalyzer,
+    Microsoft.CodeAnalysis.Testing.EmptyCodeFixProvider>;
+
+namespace Microsoft.NetCore.Analyzers.Security.UnitTests
+{
+    public class CryptographicHardwareIntrinsicsTests
+    {
+        private DiagnosticResult GetCSharpResultAt(int line, int column) =>
+            VerifyCS.Diagnostic(CryptographicHardwareIntrinsicsAnalyzer.s_rule).WithLocation(line, column);
+
+        private DiagnosticResult GetBasicResultAt(int line, int column) =>
+            VerifyVB.Diagnostic(CryptographicHardwareIntrinsicsAnalyzer.s_rule).WithLocation(line, column);
+
+        [Fact]
+        public async Task InvokingDecryptLast_CS_Diagnostic()
+        {
+            var test = new VerifyCS.Test
+            {
+                ReferenceAssemblies = new ReferenceAssemblies(
+                    "netcoreapp3.1",
+                    new PackageIdentity(
+                        "Microsoft.NETCore.App.Ref",
+                        "3.1.0"),
+                    Path.Combine("ref", "netcoreapp3.1")),
```

Comment:
:warning: I had to do it that way because `ReferenceAssemblies.NetCore.NetCoreApp31` doesn't exist. It's probably because an older version of roslyn SDK being used in 2.9.x branch.

@mavasani, Do you want to update the sdk version instead? If so, please inform me where the version should be updated.